### ISSUE TITLE
Validate counts and training controls before spending the simulation budget

### ISSUE THREAD

**Issue by pedroliman** [pedroliman/neuralsbi#45]
None of the numeric arguments on the public surface are checked. The failures show up wherever the value first reaches base R.

```r
npe(prior, sim, n_simulations = 0, density_estimator = "linear_gaussian")
#> <nsbi_npe> ... simulations: 0
#> Warning in cbind(1, x): number of rows of result is not a multiple of vector length

npe(prior, sim, n_simulations = -5)
#> Error: invalid arguments                      # stats::runif, R/prior.R:56

npe(prior, sim, n_simulations = 500, density_estimator = "mdn",
    validation_fraction = 1)
#> Error: wrong sign in 'by' argument            # seq(), R/train.R:97
```

`n_simulations = 1` succeeds silently with a meaningless fit. `validation_fraction >= 1` empties the training split so `n_tr` is `0`; `batch_size = 0` gives "invalid '(to - from)/by'"; `n_restarts = 0` skips the restart loop and surfaces as "Training failed: no restart produced a finite validation loss" (`R/train.R:157`), which blames training.

The full list of unchecked arguments:

- counts: `n_simulations`, `n` (`simulate_for_sbi`), `n_sbc`, `n_tarp`, `n_posterior_samples`, `n_rounds`, `n_init`, `n_normalization`, `n_truncation_samples`, `max_sampling_batches`, `max_proposal_batches`, `n_folds`
- training: `max_epochs`, `batch_size`, `patience`, `n_restarts`, `lr`, `validation_fraction`, `clip_grad_norm`
- architecture: `n_components`, `n_transforms`, `hidden`, `n_bins`, `tail_bound`
- MCMC: `warmup`, `thin` (tracked separately, since `thin = 0` is a live bug)

## Fix

Validate at the top of each public entry point, above any simulation, using the `check_*` helpers from the validation-layer issue. `npe()`/`nle()` already need their `match.arg()` hoisted for the same reason, so this goes in the same block.

`train_conditional_de()` also needs its own checks, since it is reachable independently and is where the arithmetic actually breaks. The one that needs thought is `validation_fraction` against `n`: the real requirement is that both splits come out non-empty, so the check belongs where `n` is known and should say what `n` would be needed rather than only that the fraction is wrong.

Error-path tests go in `test-npe.R` (new), `test-train.R` and `test-diagnostics.R`.
